components/Calendar.js

```diff
@@ -76,11 +77,14 @@ export default class Calendar extends Component {
   componentDidUpdate() {
     this.scrollToItem(VIEW_INDEX);
   }
-  
+
   componentWillReceiveProps(props) {
     if (props.selectedDate) {
       this.setState({selectedMoment: props.selectedDate});
     }
+    if (props.currentMonth) {
+      this.setState({currentMonthMoment: moment(props.currentMonth)});
+    }
```

Comment:
I think this would be more appropriate in `componentWillMount` since we only want it invoked once on initialization.